### DISCUSSION TITLE
Visual Studio Code Extension Github

### DISCUSSION THREAD

**Discussion by codefunlearn** [community/community#25441] [Packages]
<p>Hello<br>
Learner here.</p>
<ul>
<li>Using <em>Visual Studio Code</em>.</li>
<li>Working with <em>HTML</em> and <em>CSS</em>.</li>
<li>Now learning <em>Javascript</em>.</li>
</ul>
<p><strong>QUESTION</strong><br>
Looking for an <strong>extension to upload my code to Github and download my teacher’s code from within Visual Studio Code</strong>. (This was very easy in IntelliJ)</p>
<p>I have found this:</p>
<blockquote>
<p><strong>GitHub Pull Requests and Issues</strong><br>
<a href="https://marketplace.visualstudio.com/items?itemName=GitHub.vscode-pull-request-github" rel="noopener nofollow ugc">Visual Studio Code Marketplace Extensions</a></p>
</blockquote>
<p>Not sure how to use this or if this is the right way to do this.<br>
Appreciate a pointer to <strong>the best way of using Github with VSCode</strong> before I wade into the documentation.</p>
<p>Thanks!</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/codefunlearn">@codefunlearn</a>,</p>
<ol>
<li>
<p>To clone GitHub repository via VS Code, you can do like as this:<br>
</p>
<div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/a/a/aaac1dc4aebc40114dbca0387076048bc9bf36a0.gif" data-download-href="/uploads/short-url/olPVuXVzEQkHi0KuZBYjME5F4kM.gif?dl=1" title="clone-repo" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/46307996/181091149-86a3bb60-26ed-41e7-81f0-97bc7fa08802.gif" alt="clone-repo" data-base62-sha1="olPVuXVzEQkHi0KuZBYjME5F4kM" width="690" height="438" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/a/a/aaac1dc4aebc40114dbca0387076048bc9bf36a0_2_690x438.gif, https://aws1.discourse-cdn.com/github/optimized/3X/a/a/aaac1dc4aebc40114dbca0387076048bc9bf36a0_2_1035x657.gif 1.5x, https://aws1.discourse-cdn.com/github/original/3X/a/a/aaac1dc4aebc40114dbca0387076048bc9bf36a0.gif 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/a/a/aaac1dc4aebc40114dbca0387076048bc9bf36a0_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">clone-repo</span><span class="informations">1267×805 1.34 MB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
</li>
<li>
<p>To commit and push changes to GitHub repository via VS Code, you can do like as this:<br>
</p>
<div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/2/3/2318532e4a87f2076dd8d4bc45276d911ed1fd73.gif" data-download-href="/uploads/short-url/50sPircaJNY10XLw6ldROXa3cWf.gif?dl=1" title="commit-push" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/46307996/181091150-6664dec1-6a4d-44c0-9974-349b171717b2.gif" alt="commit-push" data-base62-sha1="50sPircaJNY10XLw6ldROXa3cWf" width="690" height="443" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/2/3/2318532e4a87f2076dd8d4bc45276d911ed1fd73_2_690x443.gif, https://aws1.discourse-cdn.com/github/optimized/3X/2/3/2318532e4a87f2076dd8d4bc45276d911ed1fd73_2_1035x664.gif 1.5x, https://aws1.discourse-cdn.com/github/original/3X/2/3/2318532e4a87f2076dd8d4bc45276d911ed1fd73.gif 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/2/3/2318532e4a87f2076dd8d4bc45276d911ed1fd73_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">commit-push</span><span class="informations">1210×777 1.02 MB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
</li>
</ol>

**Reply by sonivraj82** [reply]
all the files are forbidden now

**Comment by codefunlearn** [comment]
<p>Thanks, <a class="mention" href="/brightran">@brightran</a>! I am getting into the flow of things. I had just installed stuff and was looking around, now it’s making sense.<br>
Cheers</p>
